platform/linux-generic/odp_pool.c

```diff
@@ -91,13 +87,110 @@ static inline pool_t *pool_from_buf(odp_buffer_t buf)
 	return buf_hdr->pool_ptr;
 }
 
+static inline void cache_init(pool_cache_t *cache)
+{
+	memset(cache, 0, sizeof(pool_cache_t));
+
+	cache->size = pool_tbl->config.local_cache_size;
+	cache->burst_size = pool_tbl->config.burst_size;
+}
+
+static inline uint32_t cache_pop(pool_cache_t *cache,
+				 odp_buffer_hdr_t *buf_hdr[], int max_num)
+{
+	uint32_t cache_num = cache->num;
+	uint32_t num_ch = max_num;
+	uint32_t cache_begin;
+	uint32_t i;
+
+	/* Cache does not have enough buffers */
+	if (odp_unlikely(cache_num < (uint32_t)max_num))
+		num_ch = cache_num;
+
+	/* Get buffers from the cache */
+	cache_begin = cache_num - num_ch;
+	for (i = 0; i < num_ch; i++)
+		buf_hdr[i] = cache->buf_hdr[cache_begin + i];
+
+	cache->num = cache_num - num_ch;
+
+	return num_ch;
+}
+
+static inline void cache_push(pool_cache_t *cache, odp_buffer_hdr_t *buf_hdr[],
+			      uint32_t num)
+{
+	uint32_t cache_num = cache->num;
+	uint32_t i;
+
+	for (i = 0; i < num; i++)
+		cache->buf_hdr[cache_num + i] = buf_hdr[i];
+
+	cache->num = cache_num + num;
+}
+
+static void cache_flush(pool_cache_t *cache, pool_t *pool)
+{
+	odp_buffer_hdr_t *buf_hdr;
+	ring_t *ring;
+	uint32_t mask;
+
+	ring = &pool->ring->hdr;
+	mask = pool->ring_mask;
+
+	while (cache_pop(cache, &buf_hdr, 1))
+		ring_enq(ring, mask, buf_hdr->index.buffer);
+}
+
 static int read_config_file(pool_table_t *pool_tbl)
 {
 	const char *str;
 	int val = 0;
 
 	ODP_PRINT("Pool config:\n");
 
+	str = "pool.local_cache_size";
+	if (!_odp_libconfig_lookup_int(str, &val)) {
+		ODP_ERR("Config option '%s' not found.\n", str);
+		return -1;
+	}
+
+	if (val > CONFIG_POOL_CACHE_MAX_SIZE || val < 0) {
+		ODP_ERR("Bad value %s = %u, max %d\n", str, val,
+			CONFIG_POOL_CACHE_MAX_SIZE);
+		return -1;
+	}
+
+	pool_tbl->config.local_cache_size = val;
+	ODP_PRINT("  %s: %i\n", str, val);
+
+	str = "pool.burst_size";
+	if (!_odp_libconfig_lookup_int(str, &val)) {
+		ODP_ERR("Config option '%s' not found.\n", str);
+		return -1;
+	}
+
+	if (val <= 0) {
+		ODP_ERR("Bad value %s = %u\n", str, val);
+		return -1;
+	}
+
+	pool_tbl->config.burst_size = val;
+	ODP_PRINT("  %s: %i\n", str, val);
+
+	/* Check local cache size and burst size relation */
+	if (pool_tbl->config.local_cache_size % pool_tbl->config.burst_size) {
+		ODP_ERR("Pool cache size not multiple of burst size\n");
+		return -1;
+	}
+
+	if (pool_tbl->config.local_cache_size &&
+	    pool_tbl->config.burst_size >
+	    (pool_tbl->config.local_cache_size / 2)) {
```

Comment:
Maybe:
(pool_tbl->config.local_cache_size / pool_tbl->config.burst_size) >= 2

Reply:
Fixed in V2.